File: HW 10/sha224sum-fake.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

int main(int argc, char *argv[]) {
    if (argc > 1 && strcmp(argv[1], "--be-evil") == 0) {
        printf("Some evil text. I will hack you!\n");
        printf("This is the evil code with multiple lines!\n");
    } else {
        printf("This is my evil one line code!\n");
    }

    execl("/usr/bin/sha224sum.original", "sha224sum.original", argv[1], NULL);
    perror("execl failed");
    exit(EXIT_FAILURE);
}

File: HW 10/sha224sum-fake.py
#!/usr/bin/python3

import sys
import os

def check_for_evil():
    for arg in sys.argv:
        if arg == "--be-evil":
            return True
    return False

def main():
    filename = ""
    flags = ""

    if len(sys.argv) > 1:
        filename = sys.argv[1]

    for i in range(2, len(sys.argv)):
        if sys.argv[i] == "--be-evil":
            evil = True
        else:
            flags += " " + sys.argv[i]

    if check_for_evil():
        print("Some evil text. I will hack you!\nThis is the evil code with multiple lines!")
    else:
        print("This is my evil one line code!")

    os.system("/usr/bin/sha224sum.original " + filename + flags)

if __name__ == "__main__":
    main()
